cmd/kube_create.go

```diff
@@ -0,0 +1,457 @@
+package cmd
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path"
+	"text/template"
+	"time"
+
+	"github.com/cenkalti/backoff"
+	"github.com/exoscale/egoscale"
+	"github.com/gernest/wow"
+	"github.com/gernest/wow/spin"
+	"github.com/spf13/cobra"
+	"golang.org/x/crypto/ssh"
+)
+
+// kubeCreateDebug represents a debug mode flag
+var kubeCreateDebug bool
+
+// kubeBootstrapStep represents a k8s instance bootstrap step
+type kubeBootstrapStep struct {
+	name    string
+	command string
+}
+
+type kubeCluster struct {
+	Name    string
+	Version string
+	Address string
+}
+
+// kubeBootstrapSteps represents a k8s instance bootstrap steps
+var kubeBootstrapSteps = []kubeBootstrapStep{
+	{name: "Instance system upgrade", command: `\
+sudo -E DEBIAN_FRONTEND=noninteractive apt-get update && sudo -E DEBIAN_FRONTEND=noninteractive apt-get upgrade -y
+sudo -E DEBIAN_FRONTEND=noninteractive apt-get install -y \
+	apt-transport-https \
+	ca-certificates \
+	curl \
+	golang-cfssl \
+	software-properties-common
+nohup sh -c 'sleep 5s ; sudo reboot' &
+exit`},
+	{name: "Docker Engine installation", command: `\
+curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add -
+sudo add-apt-repository \
+	"deb [arch=amd64] https://download.docker.com/linux/ubuntu \
+	$(lsb_release -cs) \
+	stable"
+sudo -E DEBIAN_FRONTEND=noninteractive apt-get update
+sudo -E DEBIAN_FRONTEND=noninteractive apt-get install -y docker-ce=18.06.0~ce~3-0~ubuntu
+
+cat <<EOF > csr.json
+{
+	"hosts": ["{{ .Address }}"],
+	"key": {"algo": "rsa", "size": 2048},
+	"names": [{"C": "CH", "L": "Lausanne", "O": "Exoscale", "OU": "exokube", "ST": ""}]
+}
+EOF
+
+cfssl genkey -initca csr.json | cfssljson -bare ca
+
+cfssl gencert \
+	-ca ca.pem \
+	-ca-key ca-key.pem \
+	-hostname {{ .Address }} csr.json | cfssljson -bare
+
+cat <<EOF | sudo tee /etc/docker/daemon.json
+{
+	"hosts": ["unix:///var/run/docker.sock", "tcp://0.0.0.0:2376"],
+	"tlsverify": true,
+	"tlscacert": "/etc/docker/ca.pem",
+	"tlscert": "/etc/docker/cert.pem",
+	"tlskey": "/etc/docker/key.pem",
+	"exec-opts": ["native.cgroupdriver=systemd"],
+	"storage-driver": "overlay2",
+	"log-driver": "json-file",
+	"log-opts": {
+		"max-size": "100m"
+	}
+}
+EOF
+
+sudo mv ca.pem /etc/docker/ca.pem
+sudo mv cert.pem /etc/docker/cert.pem
+sudo mv cert-key.pem /etc/docker/key.pem
+rm -f *.{csr,json,pem}
+
+sudo sed -i -re 's#^ExecStart=.*#ExecStart=/usr/bin/dockerd#' /lib/systemd/system/docker.service
```

Comment:
It'd be nicer to use the override mechanism rather than modifying stuff in `/lib/systemd` :cowboy_hat_face: 

https://wiki.archlinux.org/index.php/Docker#Remote_API_with_systemd

Reply:
Fixed.